### PR TITLE
feat: add CoinbaseWalletProvider for solana

### PR DESCRIPTION
**Pull request by zoruka**
# Description

This PR is adding `CoinbaseWalletProvider` for allowing coinbase extension to work on Solana

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes APKT-1132
For GH issues: closes #...

# Showcase (Optional)

If there is a UI change include the screenshots with before and after state.
If new feature is being introduced, include the link to demo recording.

# Checklist

- [x] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [x] I have filled out all required sections
- [x] I have tested my changes on the preview link
- [x] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: packages/base/adapters/solana/web3js/providers/CoinbaseWalletProvider.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  provider: SolanaCoinbaseWallet
+  chains: Chain[]
+  getActiveChain: () => Chain | undefined
+}
+
+export class CoinbaseWalletProvider extends ProviderEventEmitter implements Provider {
+  public readonly name = 'Coinbase Wallet'
+  public readonly type = 'ANNOUNCED'
+  public readonly icon =
+    'data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAyNCIgaGVpZ2h0PSIxMDI0IiB2aWV3Qm94PSIwIDAgMTAyNCAxMDI0IiBmaWxsPSJub25lIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPgo8Y2lyY2xlIGN4PSI1MTIiIGN5PSI1MTIiIHI9IjUxMiIgZmlsbD0iIzAwNTJGRiIvPgo8cGF0aCBmaWxsLXJ1bGU9ImV2ZW5vZGQiIGNsaXAtcnVsZT0iZXZlbm9kZCIgZD0iTTE1MiA1MTJDMTUyIDcxMC44MjMgMzEzLjE3NyA4NzIgNTEyIDg3MkM3MTAuODIzIDg3MiA4NzIgNzEwLjgyMyA4NzIgNTEyQzg3MiAzMTMuMTc3IDcxMC44MjMgMTUyIDUxMiAxNTJDMzEzLjE3NyAxNTIgMTUyIDMxMy4xNzcgMTUyIDUxMlpNNDIwIDM5NkM0MDYuNzQ1IDM5NiAzOTYgNDA2Ljc0NSAzOTYgNDIwVjYwNEMzOTYgNjE3LjI1NSA0MDYuNzQ1IDYyOCA0MjAgNjI4SDYwNEM2MTcuMjU1IDYyOCA2MjggNjE3LjI1NSA2MjggNjA0VjQyMEM2MjggNDA2Ljc0NSA2MTcuMjU1IDM5NiA2MDQgMzk2SDQyMFoiIGZpbGw9IndoaXRlIi8+Cjwvc3ZnPgo='
```
Comment (tomiir) [inline packages/base/adapters/solana/web3js/providers/CoinbaseWalletProvider.ts:30]: where did this come from?

File: packages/base/adapters/solana/web3js/providers/CoinbaseWalletProvider.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    return signature
+  }
+
+  public async signAllTransactions<T extends AnyTransaction[]>(transactions: T): Promise<T> {
+    return (await this.provider.signAllTransactions(transactions)) as T
+  }
+
+  private getAccount<Required extends boolean>(
+    required?: Required
+  ): Required extends true ? PublicKey : PublicKey | undefined {
```
Comment (tomiir) [inline packages/base/adapters/solana/web3js/providers/CoinbaseWalletProvider.ts:106]: what is this required arg for?

**Review by tomiir — APPROVED** [review]

File: packages/base/adapters/solana/web3js/providers/CoinbaseWalletProvider.ts (hunk 1 above)
Comment (zoruka) [reply packages/base/adapters/solana/web3js/providers/CoinbaseWalletProvider.ts:30]: https://github.com/anza-xyz/wallet-adapter/blob/master/packages/wallets/coinbase/src/adapter.ts#L59-L60

File: packages/base/adapters/solana/web3js/providers/CoinbaseWalletProvider.ts (hunk 2 above)
Comment (zoruka) [reply packages/base/adapters/solana/web3js/providers/CoinbaseWalletProvider.ts:106]: to throw if not private key is not available

**Pull request merged**